src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -1583,27 +1594,29 @@ protected virtual void OnRequestTransmissionAborted(JsonRpcRequest request)
         /// <summary>
         /// Creates a dictionary which maps a request method name to its clr method name via <see cref="JsonRpcMethodAttribute" /> value.
         /// </summary>
-        /// <param name="target">Object to reflect over and analyze its methods.</param>
+        /// <param name="exposedMembersOnType">Type to reflect over and analyze its methods.</param>
+        /// <param name="target">The instance of <paramref name="exposedMembersOnType"/> to be exposed to RPC.</param>
         /// <param name="options">The options that apply for this target object.</param>
         /// <returns>Dictionary which maps a request method name to its clr method name.</returns>
-        private static Dictionary<string, List<MethodSignatureAndTarget>> GetRequestMethodToClrMethodMap(object target, JsonRpcTargetOptions options)
+        private static Dictionary<string, List<MethodSignatureAndTarget>> GetRequestMethodToClrMethodMap(TypeInfo exposedMembersOnType, object target, JsonRpcTargetOptions options)
         {
             Requires.NotNull(target, nameof(target));
+            Requires.NotNull(exposedMembersOnType, nameof(exposedMembersOnType));
             Requires.NotNull(options, nameof(options));
 
             var clrMethodToRequestMethodMap = new Dictionary<string, string>(StringComparer.Ordinal);
             var requestMethodToClrMethodNameMap = new Dictionary<string, string>(StringComparer.Ordinal);
             var requestMethodToDelegateMap = new Dictionary<string, List<MethodSignatureAndTarget>>(StringComparer.Ordinal);
             var candidateAliases = new Dictionary<string, string>(StringComparer.Ordinal);
 
-            var mapping = new MethodNameMap(target.GetType().GetTypeInfo());
+            var mapping = new MethodNameMap(exposedMembersOnType);
 
-            for (TypeInfo? t = target.GetType().GetTypeInfo(); t != null && t != typeof(object).GetTypeInfo(); t = t.BaseType?.GetTypeInfo())
+            for (TypeInfo? t = exposedMembersOnType; t != null && t != typeof(object).GetTypeInfo(); t = t.BaseType?.GetTypeInfo())
             {
                 // As we enumerate methods, skip accessor methods
                 foreach (MethodInfo method in t.DeclaredMethods.Where(m => !m.IsSpecialName))
                 {
-                    if (!options.AllowNonPublicInvocation && !method.IsPublic)
+                    if (!options.AllowNonPublicInvocation && !method.IsPublic && !exposedMembersOnType.IsInterface)
```

Comment:
Do we need a check here for static methods too?

Reply:
No. We (oddly enough) *do* support static RPC methods on target objects.